vertical-pod-autoscaler/pkg/admission-controller/resource/vpa/matcher_test.go

```diff
@@ -37,7 +40,22 @@ func parseLabelSelector(selector string) labels.Selector {
 }
 
 func TestGetMatchingVpa(t *testing.T) {
```

Comment:
Is there a testing case where VPA does not select a pod when its selector matches but the target ref does not match?

Reply:
The test was missing, I have added a test where selector match but not the ownerRef, here it is:
https://github.com/kubernetes/autoscaler/pull/6460/commits/a5cae3b59f454c6b5a801b6cc76abe65d57ee348